app/models/recipe.rb

```diff
@@ -1,3 +1,9 @@
 class Recipe < ApplicationRecord
    validates_presence_of :name, :complexity, :genre
+   has_many :recipe_ingredients
+   has_many :ingredients, through: :recipe_ingredients
+
+   def ingredient_sum
+      self.ingredients.sum(:cost)
```

Comment:
Now that you understand the use of `self` in these models, you can start dropping it if you feel comfortable! This method will work if you just do `ingredients.sum(:cost)` - try it out for fun if you'd like! There's nothing wrong with being explicit, and we always teach it that way first, but it seems like you've got a grasp of `self` now!